src/prefect/engine/task_runner.py

```diff
@@ -166,6 +166,7 @@ def initialize_run(  # type: ignore
         context.update(
             task_run_count=run_count,
             task_name=self.task.name,
+            task_run_name=self.task.task_run_name,
```

Comment:
This will work for task run names that are strings. However, we allow for dynamic task run naming based on task inputs and parameters https://github.com/PrefectHQ/prefect/blob/master/src/prefect/engine/task_runner.py#L279.

We'll need to put task run name into context after dynamic naming for this to work correctly